From c4dd72ef59770d47e40fd2cf49fed73b0a132178 Mon Sep 17 00:00:00 2001
From: Takashi Yoshii <takashi.yoshii.zj@renesas.com>
Date: Fri, 16 Nov 2012 10:52:22 +0900
Subject: [PATCH 0591/1083] serial: sh-sci: fix condition test to set SCBRR

SCBRR == 0 is valid value (divide by 1).

Signed-off-by: Takashi Yoshii <takashi.yoshii.zj@renesas.com>
Signed-off-by: Shinya Kuribayashi <shinya.kuribayashi.px@renesas.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
(cherry picked from commit 4ffc3cdb642823ebee84538addac7cde1174e314)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit afb11f52e844e38e3dd1dafa8d76e94f97ea4f99)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/tty/serial/sh-sci.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/serial/sh-sci.c b/drivers/tty/serial/sh-sci.c
index e65e546..ffeca65 100644
--- a/drivers/tty/serial/sh-sci.c
+++ b/drivers/tty/serial/sh-sci.c
@@ -1853,7 +1853,7 @@ static void sci_set_termios(struct uart_port *port, struct ktermios *termios,
 	dev_dbg(port->dev, "%s: SMR %x, t %x, SCSCR %x\n", __func__, smr_val, t,
 		s->cfg->scscr);
 
-	if (t > 0) {
+	if (t >= 0) {
 		if (t >= 256) {
 			serial_port_out(port, SCSMR, (serial_port_in(port, SCSMR) & ~3) | 1);
 			t >>= 2;
-- 
1.8.3.2

